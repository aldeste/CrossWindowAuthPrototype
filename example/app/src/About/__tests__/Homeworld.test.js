import React from "react";
import renderer from "react-test-renderer";
import Homeworld from "../Homeworld";

jest.mock("../../Tags/Tags");

const residents = [
  "Luke Skywalker",
  "C-3PO",
  "Darth Vader",
  "Owen Lars",
  "Beru Whitesun lars",
  "R5-D4",
  "Biggs Darklighter",
  "Anakin Skywalker",
  "Shmi Skywalker",
  "Cliegg Lars"
];

describe("About component", () => {
  it("Renders correctly with residents", () => {
    expect(
      renderer
        .create(<Homeworld name={"Tatooine"} residents={residents} />)
        .toJSON()
    ).toMatchSnapshot();
  });
  it("Renders correctly with empty residents", () => {
    expect(
      renderer.create(<Homeworld name={"Tatooine"} residents={[]} />).toJSON()
    ).toMatchSnapshot();
  });
});
