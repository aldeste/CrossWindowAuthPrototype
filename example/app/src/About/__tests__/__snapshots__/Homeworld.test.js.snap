// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component Renders correctly with empty residents 1`] = `
<div
  className="View__View-jelnfZ bnYeJt"
>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    You're from Tatooine
    !
  </p>
</div>
`;

exports[`About component Renders correctly with residents 1`] = `
<div
  className="View__View-jelnfZ bnYeJt"
>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    You're from Tatooine
    ! Other people are also from there! Maybe you know them?
  </p>
  <p
    className="Text__Text-jpiXUO hvuOVR"
  >
    Luke Skywalker, C-3PO, Darth Vader, Owen Lars, Beru Whitesun lars, R5-D4, Biggs Darklighter, Anakin Skywalker, Shmi Skywalker, Cliegg Lars
  </p>
</div>
`;
